java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -385,7 +385,9 @@ void executeInstruction(Instruction instruction) {
         }
         break;
       case INVOKEDYNAMIC:
-        // TODO SONARJAVA-2512
+        Instruction.InvokeDynamicInsn invokeDynamicInsn = (Instruction.InvokeDynamicInsn) instruction;
+        pop = popStack(invokeDynamicInsn.arity(), instruction.opcode);
+        programState = invokeDynamicInsn.hasReturnValue() ? pop.state.stackValue(new SymbolicValue()) : pop.state;
```

Comment:
Matter of style  : 

``` 
programState = pop.state;
if(instr.hasReturnValue() {
programState = programState.stackValue(new Sv());
 }
```

Reply:
condition removed altogether, because as mentioned above the false branch was in fact impossible